packages/flutter_driver/lib/src/common/wait.dart

```diff
@@ -0,0 +1,307 @@
+
+  /// Deserializes this command from the value generated by [serialize].
+  WaitUntilNoPendingFrame.deserialize(Map<String, String> json)
+      : super.deserialize(json);
+
+  @override
+  String get kind => 'waitUntilNoPendingFrame';
+}
+
+/// A Flutter Driver command that waits until the Flutter engine rasterizes the
+/// first frame.
+///
+/// {@template flutter.frame_rasterized_vs_presented}
+/// Usually, the time that a frame is rasterized is very close to the time that
+/// it gets presented on the display. Specifically, rasterization is the last
+/// expensive phase of a frame that's still in Flutter's control.
+/// {@endtemplate}
+///
+/// This command has been deprecated in favor of [WaitForCondition]. Construct
+/// a command that waits until no pending frame as follows:
+///
+/// ```dart
+/// WaitForCondition firstFrameRasterized = WaitForCondition(FirstFrameRasterizedCondition());
+/// ```
+@Deprecated('This command has been deprecated in favor of WaitForCondition. '
+            'Use WaitForCondition command with FirstFrameRasterizedCondition.')
+class WaitUntilFirstFrameRasterized extends Command {
+  /// Creates this command.
+  const WaitUntilFirstFrameRasterized({ Duration timeout }) : super(timeout: timeout);
+
+  /// Deserializes this command from the value generated by [serialize].
+  WaitUntilFirstFrameRasterized.deserialize(Map<String, String> json)
+      : super.deserialize(json);
+
+  @override
+  String get kind => 'waitUntilFirstFrameRasterized';
+}
+
+/// Base class for a condition that can be waited upon.
+abstract class WaitCondition {
+  /// Gets the current status of the [condition], executed in the context of the
+  /// Flutter app:
+  ///
+  /// * True, if the condition is satisfied.
+  /// * False otherwise.
+  ///
+  /// The future returned by [wait] will complete when this [condition] is
+  /// fulfilled.
+  bool get condition;
+
+  /// Returns a future that completes when [condition] turns true.
+  Future<void> wait();
+
+  /// Serializes the object to JSON.
+  Map<String, String> serialize();
+}
+
+/// Thrown to indicate a JSON serialization error.
+class SerializationException implements Exception {
+  /// Creates a [SerializationException] with an optional error message.
+  const SerializationException([this.message]);
+
+  /// The error message, possibly null.
+  final String message;
+
+  @override
+  String toString() => 'SerializationException($message)';
+}
+
+/// A condition that waits until no transient callbacks are scheduled.
+class NoTransientCallbacksCondition implements WaitCondition {
+  /// Creates a [NoTransientCallbacksCondition] instance.
+  const NoTransientCallbacksCondition();
+
+  /// Factory constructor to parse a [NoTransientCallbacksCondition] instance
+  /// from the given JSON map.
+  factory NoTransientCallbacksCondition.deserialize(Map<String, dynamic> json) {
+    if (json['conditionName'] != 'NoTransientCallbacksCondition')
```

Comment:
`assert(json != null)`

And add something to the doc string indicating that `json` must not be null.

Reply:
Alternatively, have it be `if (json == null || json['conditionName']...`

Reply:
Done.